Microsoft.DotNet.Maestro.Tasks.Tests/AddAssetTests.cs

```diff
@@ -0,0 +1,64 @@
+using FluentAssertions;
+using Microsoft.DotNet.Maestro.Client.Models;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+
+namespace Microsoft.DotNet.Maestro.Tasks.Tests
+{
+    [TestFixture]
+    public class AddAssetTests
+    {
+        [Test]
+        public void EmptyAssetList_NewAssetIsOnlyAssetInList()
+        {
+            PushMetadataToBuildAssetRegistry pushMetadata = new PushMetadataToBuildAssetRegistry();
+            List<AssetData> assetData = new List<AssetData>();
+            AssetData expectedAssetData = new AssetData(true) { Name = "testName", Version = "12345", Locations = ImmutableList<AssetLocationData>.Empty.Add(new AssetLocationData(LocationType.None) { Location = "testLocation" }) };
+
+            pushMetadata.AddAsset(assetData, expectedAssetData.Name, expectedAssetData.Version, "testLocation", LocationType.None, true);
+            assetData.Count.Should().Be(1);
+            CompareAssetData(assetData[0], expectedAssetData);
+        }
+
+        [Test]
+        public void AssetsAddedToList_NewAndPreviousAssetsInList()
+        {
+            AssetData existingAssetData = new AssetData(false)
+            {
+                Name = "ExistingAssetName",
+                Version = "56789",
+                Locations = ImmutableList<AssetLocationData>.Empty.Add(new AssetLocationData(LocationType.Container) { Location = "oldTestLocation" })
+            };
+
+            PushMetadataToBuildAssetRegistry pushMetadata = new PushMetadataToBuildAssetRegistry();
+            List<AssetData> assetData = new List<AssetData>() { existingAssetData };
+
+            AssetData newAssetData = new AssetData(true) { Name = "testName", Version = "12345", Locations = ImmutableList<AssetLocationData>.Empty.Add(new AssetLocationData(LocationType.None) { Location = "testLocation" }) };
+
+            pushMetadata.AddAsset(assetData, newAssetData.Name, newAssetData.Version, "testLocation", LocationType.None, true);
+            assetData.Count.Should().Be(2);
+            CompareAssetData(assetData[0], existingAssetData);
+            CompareAssetData(assetData[1], newAssetData);
+        }
+
+        [Test]
+        public void NullAssetList_ThrowsNullReferenceException()
+        {
+            PushMetadataToBuildAssetRegistry pushMetadata = new PushMetadataToBuildAssetRegistry();
+
+            Action act = () =>
+                pushMetadata.AddAsset(null, "testName", "12345", "testLocation", LocationType.None, true);
+            act.Should().Throw<NullReferenceException>();
```

Comment:
Shouldn't this throw an ArgumentNullException instead of a NullReferenceException?  I don't think we should ever "expect" a null ref.

Reply:
We chatted about this one and it comes down to what our code policy is for our team. There's sometimes we allow exceptions to bubble up, and in this case I felt it was allowable. Feel free to disagree though :)  My opinion was based on the fact that as this is a private method, NullRef is acceptable because the dev who is writing the code that calls it has access to this method and should validate the data before passing it along. However, if this was a public method, I feel that it's appropriate to do null checks at the beginning of the method because you don't know who's calling you and you should want to validate their data before trying to use it.

Reply:
Talked with Missy / Megan offline and we are going to leave it as is for now.